versions/latest/modules/en/pages/upgrades/roll-back.adoc

```diff
@@ -0,0 +1,160 @@
+= Rolling Back K3s
+
+You can roll back a K3s Kubernetes version after a problematic upgrade, using a combination of K3s binary downgrading and datastore restoration. This is applicable to both clusters with external databases and clusters using an embedded etcd.
+
+== Important Considerations
+
+* **Backups:** Before upgrading, ensure you have a valid database or etcd snapshot from your cluster running the older version of K3s. Without a backup, a rollback is impossible.
+* **Potential Data Loss:** The `k3s-killall.sh` script forcefully terminates K3s processes and may result in data loss if applications are not properly shut down.
+* **Version Specifics:** Always verify K3s and component versions before and after the rollback.
+
+== Rollback with External Database
+
+This section applies to K3s clusters using an external database (e.g., PostgreSQL, MySQL).
+
+. If the cluster is running and the Kubernetes API is available, gracefully stop workloads by draining all nodes:
++
+[,bash]
+----
+kubectl drain --ignore-daemonsets --delete-emptydir-data <NODE-ONE-NAME> <NODE-TWO-NAME> <NODE-THREE-NAME> ...
+----
++
+[NOTE]    
+====
+This process may disrupt running applications.
+====
+
+. On each node, stop the K3s service and all running pod processes:
++
+[,bash]
+----
+k3s-killall.sh
+----
+
+. Restore a database snapshot taken before upgrading K3s. For example, if you're using PostgreSQL, run the following command: 
++
+[,bash]
+----
+pg_restore -U <DB-USER> -d <DB-NAME> <BACKUP_FILE>
+----
+Verify the integrity of the restored database.
```

Comment:
Is `Verify the integrity...` a subparagraph of the `Restore a database...` step or is it intended to be its own step? This line needs to be updated in either case.